instrumentation/zio/zio-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/zio/v2_0/FiberContext.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.zio.v2_0;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import javax.annotation.Nullable;
+
+public class FiberContext {
+
+  private Context context;
+  @Nullable private Scope scope;
+
+  private FiberContext(Context context) {
+    this.context = context;
+    this.scope = null;
+  }
+
+  public static FiberContext create() {
+    return new FiberContext(Context.current());
+  }
+
+  public void onEnd() {
```

Comment:
I would use
```java
  public void onEnd() {
    if (this.scope != null) {
      this.scope.close();
      this.scope = null;
    }
  }

  public void onSuspend() {
    this.context = Context.current();

    if (this.scope != null) {
      this.scope.close();
      this.scope = null;
    }
  }

  public void onResume() {
    this.scope = this.context.makeCurrent();
  }
```

Reply:
With this implementation tests fail for me. Maybe you've made some other changes as well? Also I've added more tests in my previous commits, so maybe you haven't pulled the latest version?

Reply:
Thanks, you are correct I forgot to pull the latest version.